src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -390,31 +395,79 @@ public static void Register(Type t, DeepCopier cop, Serializer ser, Deserializer
                 {
                     if (cop != null)
                     {
-                        lock (copiers)
+                        if (_setupDone)
                         {
-                            DeepCopier current;
-                            if (forceOverride || !copiers.TryGetValue(t.TypeHandle, out current) || (current == null))
+                            lock (copiers)
```

Comment:
what is the purpose of the double locking (on `registeredTypes` and then individually on `copiers`)? Is some other method modifying the collections individually?